scraper/src/youtube2zim/scraper.py

```diff
@@ -1108,3 +1120,240 @@ def to_data_js(video):
+                thumbnail_path=get_thumbnail_path(
+                    videos[0]["contentDetails"]["videoId"]
+                ),
+            )
+
+        def generate_playlist_preview_object(playlist):
+            videos = get_videos_list(playlist)
+            slug = get_playlist_slug(playlist)
+            return PlaylistPreview(
+                slug=slug,
+                id=playlist.playlist_id,
+                title=playlist.title,
+                thumbnail_path=get_thumbnail_path(
+                    videos[0]["contentDetails"]["videoId"]
+                ),
+                count=len(videos),
+            )
+
+        def get_playlist_slug(playlist):
+            return f"{get_slug(playlist.title)}-{playlist.playlist_id[-4:]}"
+
+        videos = load_mandatory_json(self.cache_dir, "videos").values()
+        # filter videos so we only include the ones we could retrieve
+        videos = list(filter(is_present, videos))
+        videos_channels = load_mandatory_json(self.cache_dir, "videos_channels")
+        has_channel = functools.partial(video_has_channel, videos_channels)
+        # filter videos to exclude those for which we have no channel (#76)
+        videos = list(filter(has_channel, videos))
+        for video in videos:
+            slug = get_video_slug(video)
+            self.zim_file.add_item_for(
+                path=f"videos/{slug}.json",
+                title=slug,
+                content=generate_video_object(video).model_dump_json(
+                    by_alias=True, indent=2
+                ),
+                mimetype="application/json",
+                is_front=False,
+            )
+
+        # write playlists JSON files
+        playlist_list = []
+
+        main_playlist_slug = None
+        if len(self.playlists) > 0:
+            main_playlist_slug = get_playlist_slug(
+                self.playlists[0]
+            )  # set first playlist as main playlist
+
+        for playlist in self.playlists:
+            playlist_slug = get_playlist_slug(playlist)
+            playlist_path = f"playlists/{playlist_slug}.json"
+
+            if playlist.playlist_id != self.uploads_playlist_id:
+                playlist_list.append(generate_playlist_preview_object(playlist))
+            else:
+                main_playlist_slug = (
+                    playlist_slug  # set uploads playlist as main playlist
+                )
+
+            self.zim_file.add_item_for(
+                path=playlist_path,
+                title=playlist.title,
+                content=generate_playlist_object(playlist).model_dump_json(
+                    by_alias=True, indent=2
+                ),
+                mimetype="application/json",
+                is_front=False,
+            )
+
+        # write playlists.json file
+        self.zim_file.add_item_for(
+            path="playlists.json",
+            title="Playlists",
+            content=json.dumps(playlist_list, default=pydantic_encoder, indent=2),
```

Comment:
Rather than using this different method (everywhere else you use `model_dump_json`) I would rather add a `Playlists` class to the schemas, with for now only one property named `items` which is the list of playlist.

This avoid the specifity, allows for greater possibility of extension (should we need more than simply the list of playlists in this file) and avoids security issues (I'm not a specialist, but it looks like it is not recommended to have a list as JSON file content).

Reply:
Changed in [25b7546](https://github.com/openzim/youtube/pull/219/commits/25b7546cc4a3657c357a8ce3e8acb2fb28c90439)